### ISSUE TITLE
Comma key of numeric keypad creates dot instead of comma in transaction amount field

### ISSUE THREAD

**Issue by schoch** [moneymanagerex/moneymanagerex#6441]
MMEX version:
 - [ ] 1.6.4
 - [x] 1.7.0
 - [ ] 1.7.1
 - [ ] 1.7.2
 - [ ] Other (please specify)

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

I have a laptop with a full-size keyboard. This means that I have two keys to generate a comma. One of them is part of the numeric keypad. When I use this specific key to add an amount of money to a transaction, it produces an unexpected result. The key generates in this very specific situation a dot and not a comma. And that leads to wrong amounts since the dot ist used as a thousands separator in my locale (German). The comma key works as expected in every other input field - even in all input fields in the MMEX application. The other comma key works as expected in every situation.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

- Create new transaction
- Fill in any amount of money which contains values before and after the decimal point
- use the comma key of the numeric keypad as a separator
- this specific comma key creates an dot but not a comma
- this dot is interpreted as a thousands separator and that leads to a wrong amount

Expected result:

- Comma key of the numeric keypad should add a comma but not a dot

Actual result:

- Comma key of the numeric keypad generates a dot

# Additional information

The comma key (of the numeric keypad) generates a comma in all other input field. And that includes all input fields in the MMEX application. And outside of MMEX it works also as expected.

**System info:**

- Windows 11 pro
- German Version
- Locale: German (German)

**Comment by renato-mmex** [comment]
and if you use this setting here
<img width="386" alt="image" src="https://github.com/moneymanagerex/moneymanagerex/assets/76920138/03dab17e-4f49-4a8c-9337-c8edb84c0d2e">

**Comment by schoch** [comment]
Yes, I have exactly that setting set to the German format. And it works - this ist not the actual problem.

My laptop has two different keys for the comma symbol. And one of these keys does not work as expected but only in the amount input field. It works in every other app on my laptop and it works in every other input field in the MMEX app. I am a dev by myself and I can't explain that strange behavior to me.

![IMG_2487](https://github.com/moneymanagerex/moneymanagerex/assets/154128/b74e7056-b80d-4208-a90f-da6c1b6506eb)

**Comment by renato-mmex** [comment]
I have already understood the problem, I am just trying to find the cause.
From my point of view, there are two possibilities:
a) Setting in Windows https://www.netzwelt.de/anleitung/180284-windows-10-punkt-statt-komma-nummernblock-festlegenso-gehts.html
b) Error in the code table of de-DE
by the way: when I activate de-DE with my Windows 11, I also have the point, so probably b)

**Comment by schoch** [comment]
Thanks for your effort. I would also go for b)

When I searched the web for possible solutions, I found a) and verified that setting.

Please let me know if I can provide any more information.

**Comment by renato-mmex** [comment]
There was the same problem before: https://github.com/moneymanagerex/moneymanagerex/issues/3622
@whalley

**Comment by whalley** [comment]
We seem to have some code to process the Keypad 'decimal point' key key and it looks to always map the input to the DECIMAL POINT in the currency definition... Maybe some bug in this code.

https://github.com/moneymanagerex/moneymanagerex/blob/5e783381f024acd49e53b6b2753efd233c009251/src/validators.h#L83-L95

**Comment by n-stein** [comment]
> We seem to have some code to process the Keypad 'decimal point' key key and it looks to always map the input to the DECIMAL POINT in the currency definition

I changed this behavior in the fix for #6474. It should now map to the locale decimal (if locale is set in Options), otherwise it will map to the currency decimal. Maybe that resolves this issue as well.
